docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,172 @@
+developed using the same logic.
+
+This makes the CLI and associated documentation not super intuitive and it
+also makes the maintenance more expensive in the long term.
+
+The goal of the project is to provide MetalK8s administrator with an intuitive
+and easy to use set of tools in order to administrate and operate a finite set
+of functionalities.
+
+Because kubectl is already in place and is well known by Kubernetes
+administrators, it will be used as a standard to follow for all other MetalK8s
+CLIs:
+
+- CLI follows kubectl style: kubectl <action> <resource>
+- CLI provides an exhaustive help, per action, with relevant examples
+- CLI provides <action> help when the command is not valid
+- CLI is not interactive (except if password input is needed)
+- CLI should not require password input
+- CLI provides a dryrun mode for intrusive operations
+- CLI provides a verbose (or debug) mode
+- CLI implementation relies on secure APIs
+- CLI support <action> completion for easy discovery
+- CLI output is standardized and human readable by default
+- CLI output can be formatted in JSON or YAML
+- CLI can be executed from outside of the cluster (except maybe for very first
+  install script)
+
+When it is possible, it would make sense to leverage kubectl plugin
+
+All functionalities are exposed through 2 distincts CLI:
+- kubectl (enriched using kubectl approach)
+- mk8sctl: a new CLI, exposing specific MetalK8s functionalities
+
+In order to operate the cluster with mk8sctl from outside of the cluster, a
+specific pkg (for each OS) or a procedure explaining how to deploy it is
+available.
+The mk8scli is deployed and available by default on bootstrap nodes
+
+Requirements
+------------
+
+Not listing all commands that are already available through kubectl.
+Only describing commands that are missing or commands that can be simplified
+using new command line arguments and predefined manifest.
+
+
+Cluster Resources Administration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**tool: kubectl**
+
++------------+------------------+-----------------------------------------+
+| Resource   | action           | parameters                              |
++============+==================+=========================================+
+| node       | add              | name, ssh-user, hostname or ip,         |
+|            |                  | ssh port, ssh-key-path, sudo-required,  |
+|            |                  | roles, manifest-template-path           |
++------------+------------------+-----------------------------------------+
+| the manifest-template-path is the path to the manifest describing the   |
+| resource. Everything that is in the command line, override the manifest |
+| content. This apply to Node and Volumes.                                |
++------------+------------------+-----------------------------------------+
+| node       | edit             | name, hostname (to change node hostname)|
```

Comment:
What would 'changing the hostname' do? You can't change the `name` of an object in K8s...

Reply:
@NicolasT actually it will do all of this: https://metal-k8s.readthedocs.io/en/latest/operation/changing_node_hostname.html

Reply:
Oh, that's not "changing a hostname". This procedure is basically replacing a Node entirely (it's not even draining it beforehand). Not sure we want to expose this as a command, especially not one that makes you believe it's "just changing a value in some YAML".

Reply:
will remove this one